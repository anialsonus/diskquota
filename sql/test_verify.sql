CREATE SCHEMA s1;

SELECT diskquota.set_schema_quota('s1', '1MB');

CREATE TABLE s1.a(i int);

INSERT INTO s1.a SELECT generate_series(1,100);

-- overflow, but not detected yet
INSERT INTO s1.a SELECT generate_series(1,10000000);

SELECT pg_sleep(5);

INSERT INTO s1.a SELECT generate_series(1,100);

SELECT diskquota.set_schema_quota('s1', '-1');

SELECT pg_sleep(5);
INSERT INTO s1.a SELECT generate_series(1,100);

DROP SCHEMA s1 CASCADE;
