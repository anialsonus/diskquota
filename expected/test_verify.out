CREATE SCHEMA s1;
SELECT diskquota.set_schema_quota('s1', '1MB');
 set_schema_quota 
------------------
 
(1 row)

CREATE TABLE s1.a(i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO s1.a SELECT generate_series(1,100);
-- overflow, but not detected yet
INSERT INTO s1.a SELECT generate_series(1,10000000);
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

INSERT INTO s1.a SELECT generate_series(1,100);
ERROR:  schema's disk space quota exceeded with name:s1
SELECT diskquota.set_schema_quota('s1', '-1');
 set_schema_quota 
------------------
 
(1 row)

SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

INSERT INTO s1.a SELECT generate_series(1,100);
DROP SCHEMA s1 CASCADE;
NOTICE:  drop cascades to table s1.a
